src/systemdhook.c

```diff
@@ -382,7 +387,18 @@ static int move_mounts(const char *id,
 	snprintf(mount_dir, PATH_MAX, "%s%s", rootfs, path);
 
 	/* Create a temporary directory to move the PATH mounts to */
-	char temp_template[] = "/tmp/ocitmp.XXXXXX";
+	char temp_template[] = "/run/oci/ocitmp.XXXXXX";
+	_cleanup_umount_ char *tempdir = "/run/oci";
+
+	if ((makepath(tempdir, 0755) == -1) && (errno != EEXIST)) {
+		return -1;
+	}
+	if (bind_mount(id, tempdir, tempdir, false) == -1) {
```

Comment:
This looks like it's going to re-bind `/run/oci` every time we use this hook for a new container.  If this needs to happen on `tempdir` and not on `tmp_dir` below, can we add a guard and skip this and the following `MS_PRIVATE|MS_REC` call if `tempdir` is already `MS_PRIVATE|MS_REC`?  You might need a flock somewhere to avoid racing between the check and the guarded `bind_mount` and `mount` calls.